pkg/funcs/functions.go

```diff
@@ -0,0 +1,85 @@
+package pkg
+
+import (
+	"time"
+
+	"github.com/cloudprivacylabs/lsa/pkg/ls"
+	oc "github.com/cloudprivacylabs/opencypher"
+	"github.com/neo4j/neo4j-go-driver/neo4j"
+)
+
+func init() {
+	oc.RegisterGlobalFunc(oc.Function{
+		Name:      "age",
+		MinArgs:   1,
+		MaxArgs:   2,
+		ValueFunc: ageFunc,
+	})
+	oc.RegisterGlobalFunc(oc.Function{
+		Name:      "lookupValueset",
+		MinArgs:   1,
+		MaxArgs:   -1,
+		ValueFunc: lookupValuesetFunc,
+	})
+}
+
+func ageFunc(ctx *oc.EvalContext, args []oc.Value) (oc.Value, error) {
+	if args[0].Get() == nil {
+		return oc.RValue{}, nil
+	}
+	goDate := time.Time{}
+	// check if argument a neo4j.Date or neo4j.LocalDateTime
+	if date, ok := args[0].Get().(neo4j.Date); ok {
+		goDate = date.Time()
+	} else if dateTime, ok := args[0].Get().(neo4j.LocalDateTime); ok {
+		goDate = dateTime.Time()
+	}
+	tm := time.Now()
+	// check for second argument type, if null default to time.Now()
+	if len(args) > 1 && args[1].Get() != nil {
+		if date, ok := args[1].Get().(neo4j.Date); ok {
+			tm = date.Time()
+		} else if dateTime, ok := args[1].Get().(neo4j.LocalDateTime); ok {
+			tm = dateTime.Time()
+		}
+	}
+	// determine age by total month difference, days will be tiebreak
+	deltaMonths := (tm.Year()*12 + int(tm.Month()) - 1) - (goDate.Year()*12 + int(goDate.Month()) - 1)
+	if goDate.Day() >= tm.Day() {
+		age := deltaMonths / 12
+		return oc.RValue{Value: age}, nil
+	}
+	deltaMonths--
+	age := deltaMonths / 12
+	return oc.RValue{Value: age}, nil
+}
+
+var ValuesetLookupFunc func(*ls.Context, ls.ValuesetLookupRequest) (ls.ValuesetLookupResponse, error)
+
+func lookupValuesetFunc(ctx *oc.EvalContext, args []oc.Value) (oc.Value, error) {
+	if args[0] == nil {
+		return oc.RValue{}, nil
+	}
+	vs, ok := args[0].Get().(map[string]oc.Value)
+	if !ok {
+		return oc.RValue{}, nil
+	}
+	reqs := make([]ls.ValuesetLookupRequest, 0)
+	for k, v := range vs {
+		reqs = append(reqs, ls.ValuesetLookupRequest{
+			TableIDs:  []string{k},
```

Comment:
TableIDs comes from []string{vs["tableId"]}
Rest of the keys in that map will go to KeyValues
You cannot type assert v.Get() to map[string]string.